base/process.jl

```diff
@@ -137,7 +137,7 @@ end
         iolock_end()
     end
     if err != 0
-        throw(_UVError("could not spawn " * repr(cmd), err))
+        throw(_UVError("could not spawn `" * cmd[1] * '`', err))
```

Comment:
I think this is already an improvement, but I think we could go one step further and not include the "or directory" bit here, by special casing `ENOENT` and writing our own custom message, then invoking IOError directly, rather than using the `_UVError` function:

```suggestion
        if err == UV_ENOENT
            throw(IOError("could not spawn `" * cmd[1] * '`' * ": no such file (ENOENT)"))
        else
            throw(_UVError("could not spawn `" * cmd[1] * '`', err))
        end
```

The motivation is so the message says e.g.
```
IOError: could not spawn `start`: no such file (ENOENT)
```
rather than
```
IOError: could not spawn `start`: no such file or directory (ENOENT)
```
where the "or directory" can be confusing, since generally one cannot spawn a directory anyway.

Reply:
If we want to be accurate here, the thing that failed was `file`, which is only usually equal to `cmd[1]` by convention
```suggestion
        throw(_UVError("could not spawn " * repr(file), err))
```

Reply:
> where the "or directory" can be confusing, since generally one cannot spawn a directory anyway.

FWIW, the "or directory" part is in reference to the fact that the command could be a path, and the error is thrown if any of the directories in that path are missing (where the full path is potentially being formed by looking through PATH or using cwd). The error for attempting to spawn a folder is typically EACCES:

```
julia> run(`./base/`)
ERROR: IOError: could not spawn `./base/`: permission denied (EACCES)
```

Reply:
Maybe we can say "path not found"?